### ISSUE TITLE
backward question

### ISSUE THREAD

**Issue by nandiya** [thehemen/cnn-from-scratch#1]
https://github.com/thehemen/cnn-from-scratch/blob/9a0947d51483be9757a29dfeae399bcae46818a3/src/fully_connected_layer.cpp#L66

if i may know, why variable "outputDerVec" wont be used again after keeping the vector of softmax derivative. lets say, when we call the backward function in fully connected layer the activation function is softmax, after calculating the backprop of softmax and get the vector result, i dont see the code keep the value in class or return the value or any used in this variable. Please tell me what its used for after that, becs in theory we will use each value in the vector to upgrade all related weights in all layers.

**Comment by thehemen** [comment]
Yes, that's strange... I've forgotten to remove this variable (outputDerVec) or just to use it later.
However, I'm not strong in backpropagation theory, so I can't add anything yet.

**Comment by nandiya** [comment]
i see.. 
is that mean this program still not finished yet? bcs i havent tried to run it and i'm still trying to understand the code and the flow

**Comment by thehemen** [comment]
Actually, yes, it's not finished. I've written it "just for training purposes", not for any kind of serious projects.
